core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java

```diff
@@ -47,9 +50,22 @@
  * database.
  */
 public class MssqlSqlDialect extends SqlDialect {
+    /**
+     * Mssql type system.
+     */
+  public static final RelDataTypeSystem MSSQL_TYPE_SYSTEM =
+            new RelDataTypeSystemImpl() {
+      @Override public int getDefaultPrecision(SqlTypeName typeName) {
+          if (typeName == SqlTypeName.CHAR) {
+          return RelDataType.PRECISION_NOT_SPECIFIED;
```

Comment:
indentation is off

Reply:
Ok, thank you for helping me review the code, I will modify it